### PR TITLE
cwrapper: symbol_set takes a const char *

### PR DESCRIPTION
**Pull request by abinashmeher999**

### CODE REVIEW

File: symengine/cwrapper.cpp [hunk 1]
```diff
@@ -160,7 +160,7 @@ TypeID basic_get_type(const basic s)
     return static_cast<TypeID>(s->m->get_type_code());
 }
 
-void symbol_set(basic s, char *c)
+void symbol_set(basic s, const char *c)
```
Comment (isuruf) [inline symengine/cwrapper.cpp]: Use `const char * const c` here

File: symengine/cwrapper.h [hunk 2]
```diff
@@ -112,7 +112,10 @@ TypeID basic_get_class_id(const char *c);
 char *basic_get_class_from_id(TypeID id);
 
 //! Assign to s, a symbol with string representation c.
-void symbol_set(basic s, char *c);
+//! This function creates a new SymEngine::Symbol from a copy of
+//! the string in c, thus the caller is free to use c afterwards,
+//! and also the caller must free c.
```
Comment (isuruf) [inline symengine/cwrapper.h]: This description is not needed. It can be deduced from the argument type.

File: symengine/cwrapper.cpp (hunk 1 above)
Comment (certik) [inline symengine/cwrapper.cpp]: Uff, is that a const pointer to a constant memory?

Comment (certik) [inline symengine/cwrapper.cpp]: We currently don't use `const` in cases like this one:
```
void integer_set_si(basic s, long i)
```
which should probably be instead:
```
void integer_set_si(basic s, const long i)
```
So then in the same way, we could use
```
void symbol_set(basic s, const char * const c)
```

Comment (isuruf) [inline symengine/cwrapper.cpp]: Now that I think about it, `const char *` is enough. It doesn't matter if the pointer is assigned to or not as long as the memory pointed to is not changed.

File: symengine/cwrapper.h (hunk 2 above)
Comment (abinashmeher999) [inline symengine/cwrapper.h]: Done.

File: symengine/cwrapper.cpp (hunk 1 above)
Comment (abinashmeher999) [inline symengine/cwrapper.cpp]: I was under that impression that I can't pass a `const <type>` to a function that takes `<type>` which is why I sent this PR. Turns out I was wrong and I can indeed do it.

File: symengine/cwrapper.h (hunk 2 above)
Comment (isuruf) [inline symengine/cwrapper.h]: I meant the last line about freeing the memory. Otherwise the comment is good

Comment (abinashmeher999) [inline symengine/cwrapper.h]: Ah, I see. Fixed it.

File: symengine/cwrapper.cpp (hunk 1 above)
Comment (certik) [inline symengine/cwrapper.cpp]: @isuruf exactly. That's why I think that `const char *` is enough. It's more readable and it does what we need it to do.

@abinashmeher999 the const is useful to check *your* code (not the user's code) that it doesn't accidentally modify it.

**Comment by certik** [comment]
That's fine.

**Pull request merged**